doc/port-add-del-dynamically/dynamic_port_add_del_hld.md

```diff
@@ -0,0 +1,305 @@
+-   Line card manager will add the buffer configuration for this port through a default buffer cfg template.
+-   Line cart manager will enable the port.
+
+•   Pg_profile_lookup file has values that will be used for static buffer configuration.
+for each port speed and cable length we have buffer size value, xon and xoff value <br />
+For example:
+
+
+|speed cable | cable | size  |  xon  |  xoff |  threshold |
+|:----------:|:-----:|:-----:|:-----:|:-----:|:----------:|
+|  10000     | 5m    | 49152 | 19456 | 29696 |      0     |
+|  25000     | 5m    | 49152 | 19456 | 29696 |      0     |
+|  40000     | 5m    | 49152 | 19456 | 29696 |      0     |
+|  50000     | 5m    | 49152 | 19456 | 29696 |      0     |
+
+On the line-card system we will use different types of line cards (maybe with different gearboxes), the values on the pg_profile_lookup will be used for all the types.<br />
+we may need to consider using pg_profile_lookup.ini for each line card type.<br />
+•   When port is added to the config db – the speed and the admin state is saved on internal db<br />
+•   After port was added the user can add buffer configuration to this port (dynamic or static configuration) and only then the buffermgr will set the buffer configuration on App table<br />
+
+•   We have rare situation of race condition in the add port flow:<br />
+
+ ![possible buffermgr race condition](images/buffermgr_possible_race.png)
+
+
+
+##### Del port:
+•   Before removing a port all buffer configuration needs to be removed<br />
+
+We have also possible way for race condition:<br />
+
+ ![possible buffermgr delete port race condition](images/buffermgr_possible_delete_race.png)
+
+•   If the portsyncd is “quicker” than the buffermgr the orchagent will try to remove the port from SAI before the buffer configuration was removed.<br />
+•   Need to test this scenario in order to check if this race condition is reproducing or it’s rare scenario<br />
+•   Solution for this: <br />
+Need to add to orchagent the ability to add the buffer configuration of a port and increase a reference counter for each port, in the same way ACL cfg on port is working. We already have infrastructure for this just need to add the buffer cfg to use it. If a port has with buffer cfg on – this port will not be removed.
+
+
+** Buffer changes PR: ** <br />
+[https://github.com/Azure/sonic-swss/pull/2022](https://github.com/Azure/sonic-swss/pull/2022)
+
+
+
+#### LLDP – lldpmgrd – implementation today:
+•   Add port: receive port entry set on port config db -> check if oper state is up or wait until oper state up event is received from app db  add lldp port entry  with lldpcli command
+•   Del port: when host interface is removed from system lldp configuration is removed also.
+ 
+
+
+
+
+ ![Add port - LLDP- current implementation](images/lldp_before.png)
+
+
+
+
+
+Suggested change:
```

Comment:
Please clarify what problem you are trying to address here.

Reply:
i will add description for it